src/components/FilterCollapse.js

```diff
@@ -445,7 +445,7 @@ class FilterCollapse extends Component {
   }
 
   async updateFilters(overrideFilter) {
-    if (Query.get('f') === this.state.filterInput) {
+    if (this.props.filter != null && Query.get('f') === this.state.filterInput) {
```

Comment:
If the filter is initially unset, its value will be `undefined`, not `null`. This check will still work correctly because `!=` coerces the two, but it can be confusing. I also think we want to avoid coercing equality operators (`==` and `!=`) where possible. A simpler way to do this is to rely on the fact that `undefined` and `null` are both falsy, while a defined parser is truthy, which removes the need for the comparison entirely.

```suggestion
    if (this.props.filter && Query.get('f') === this.state.filterInput) {
```